### PR TITLE
fix: enum option issues

### PR DESCRIPTION
**Pull request by NeloBlivion**
## Summary

Makes two changes for better enum support in options:
- If an enum's docstring is greater than 100 characters, truncate it (fixes issues on Python 3.11)
- If an enum has more than 25 values, use `basic_autocomplete` instead of `choices` and forces the type back to string.

Related thread: https://discord.com/channels/881207955029110855/1238137800440942684/1238624044396314714

## Information

- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

**Comment by JustaSqu1d** [comment]
It would be nice if this behavior was documented somewhere

**Comment by Lulalaby** [comment]
What squid said

File: discord/commands/options.py [hunk 1]
```diff
@@ -196,6 +197,8 @@ def __init__(
         if input_type_is_class and issubclass(input_type, (Enum, DiscordEnum)):
             if description is None:
                 description = inspect.getdoc(input_type)
```
Comment (Icebluewolf) [inline discord/commands/options.py:199]: What is the point of using `getdoc`? I think `cleandoc` is a better choice here. `getdoc` calls `cleandoc` but it also inherits docstrings from parents if the class does not have one. There is no reason that someone would want the description of their command to be the docstring for `Enum` or `DiscordEnum`.
```suggestion
                description = inspect.cleandoc(input_type.__doc__)
```

**Review by Icebluewolf — CHANGES_REQUESTED** [review]

**Comment by Dorukyum** [comment]
@NeloBlivion see suggestions

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**